### PR TITLE
js-fundamentals-mini-challenge

### PR DESCRIPTION
**Pull request by DkFresh89**
Love me some feedback!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,88 +1,122 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
+console.log("%cQuestion 1", "color: red") 
 
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+let counter = 1
```
Comment (lindsayMecher) [inline index.js:5]: Nice :100: `let` allows us to reassign our variables

File: index.js [hunk 2]
```diff
@@ -1,88 +1,122 @@
...
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
-
-// console.log(name)
+console.log("%cQuestion 2", "color: red") 
+const name = "Doug"
+console.log(name)
```
Comment (lindsayMecher) [inline index.js:20]: Great! The key here was using `const` because it doesn't allow us to reassign our variable

File: index.js [hunk 3]
```diff
@@ -1,88 +1,122 @@
...
-// // => 11
-// console.log("%c----------", "color: red") 
+function drinkWater(currentThirstLevel){
+  console.log("Man I sure am thirsty")
+  currentThirstLevel --
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel
+}
+
+console.log(drinkWater(12))
```
Comment (lindsayMecher) [inline index.js:40]: Great! Could be further simplified to:

```
function drinkWater(thirstLevel) {
  console.log("Man I sure am thirsty");
  console.log("Ahh that hits the spot");
  return thirstLevel -= 1;
}
```

File: index.js [hunk 4]
```diff
@@ -1,88 +1,122 @@
...
+  if (num === maybeNum){
+    return "same same"
+  }
+  if (num == maybeNum){
+    return "same same (but different)"
+  }
+  else 
+  return "different"
+}
+
```
Comment (lindsayMecher) [inline index.js:62]: Good! I would alter the syntax in the following way to keep with convention:

```
function sameSameButDifferent(num, maybeNum) {
  if (num === maybeNum) {
    console.log("same same")
  } else if (num == maybeNum) {
    console.log("same same but different")
  } else {
    console.log("different")
  }
}
```

File: index.js [hunk 5]
```diff
@@ -1,88 +1,122 @@
...
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%cQuestion 5", "color: red")
+const student1 = { name: "Duane", grade: 88 }
+function updateGrade(student, grade){
+  student.grade = grade
```
Comment (lindsayMecher) [inline index.js:83]: great!

File: index.js [hunk 6]
```diff
@@ -1,88 +1,122 @@
...
+]
+
+function printNameAndPhones(users){
+  users.forEach(function(value){
+    console.log(value.name)
+    console.log(`Cell: ${value.phones.cell}`)
+    console.log(`Office: ${value.phones.office}`)
+  })
+}
+printNameAndPhones(users)
```
Comment (lindsayMecher) [inline index.js:112]: Great! Love that you're using forEach

File: index.js [hunk 7]
```diff
@@ -100,23 +134,34 @@ function myMap(array, callback) {
...
+function greet(name){
+  return `Hi, ${name}!`
+}
 
-// console.log(myMap(["Raffy", "Chase"], greet)) 
-// // => ["Hi, Raffy!", "Hi, Chase!"]
-// console.log("%c----------", "color: red") 
+console.log(myMap(["Raffy", "Chase"], greet)) 
+// => ["Hi, Raffy!", "Hi, Chase!"]
+console.log("%c----------", "color: red")
```
Comment (lindsayMecher) [inline index.js:164]: Good! You just need to define the myMap function:

```
function myMap(arr, fn) {
  return arr.map(function(item){
    return fn(item);
  });
}
```

File: index.js [hunk 8]
```diff
@@ -136,49 +181,65 @@ function nowServing(line) {
...
+let ticketNumber = 0
 
 // ***** Scope & Closures - Question 1 *****
 
+function takeATicketNumber(line){
+  ticketNumber++
+  line.push(ticketNumber)
+  return `Welcome. You are ticket number ${ticketNumber}`
+}
+
```
Comment (lindsayMecher) [inline index.js:193]: great!

File: index.js [hunk 9]
```diff
@@ -136,49 +181,65 @@ function nowServing(line) {
...
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc() {
+  let newNum = 0
+  return function(line) {
+  newNum++
+  line.push(newNum)
+  return `Welcome. You are ticket number ${newNum}`
+}
+}
```
Comment (lindsayMecher) [inline index.js:226]: awesome!

**Review by lindsayMecher — COMMENTED** [review]
Great work!!!

File: index.js (hunk 7 above)
Comment (DkFresh89) [reply index.js:164]: I thought the myMap function was written out for us already?  Would you be able to elaborate?

Comment (lindsayMecher) [reply index.js:164]: yes you are right. Sorry about that!

**Pull request closed without merging**